### ISSUE TITLE
Remove Buffer in favor of a base64 encoding library

### ISSUE THREAD

**Issue by aljones15** [Jxck/structured-field-values#9]
The use case:

The latest version of [`Signing HTTP Signatures`](https://www.ietf.org/archive/id/draft-ietf-httpbis-message-signatures-04.html#name-the-signature-input-http-he) uses structured fields. HTTP Signatures are regularly created client side, meaning that users of any javascript implementation of the spec will need a way of serializing `Signature-Input` and `Signature` to structured field dictionaries.

The problem is in this snippet:

https://github.com/Jxck/structured-field-values/blob/26ddd6e5ab16d1043d8706f5ed9abe6a64d71e40/index.js#L1448-L1458

Uses `Buffer` which is node only. This means that when I want to use structured fields in headers for client side requests I must include a `Buffer` polyfill which adds around 200 kb to an implementation. Numerous cross environment libraries do base 64 encoding on `Uint8Array`s so this seems like something that is easily fixable. Another option is to use browser aliases in your `package.json` with a different base64 encoding method for browsers and node. Basically using either an npm package that works in node and the browser or a browser alias is smaller than the polyfill's build tools like webpack and rollup will add for you.

**Comment by Jxck** [comment]
This caused because Node doesn't have `atob` `btoa`.
But in latest node has it.
https://nodejs.org/en/blog/release/v16.0.0/
So I'll remove Buffer here.

**Comment by aljones15** [comment]
I would recommend a browser alias in the `package.json` then, but that does make your library node >= 16 only which comes with it's own issues.

**Comment by Jxck** [comment]
then I think below make Buffer to plain object and reduce a bundled js.
It doesn't requires adding dependencies I need to keep care.

```js
"browser": {
  "buffer": false
}
```

and also I'll fixing feature detection based on `atob` existence not `window`

```js
if (globalThis.atob) {
} else {
}
```

is that works for your use-case ?

**Comment by aljones15** [comment]
I believe that should be:

```js
"browser": {
  "Buffer": false
}
```

Sorry for the late reply (I replied to this via email and github dropped the message) your proposed solution looks great to me. My use case is node >= 14 and firefox and chrome as supported browsers. Not sure on which versions of firefox and chrome, but probably the last 5 years? BTW I saw `serializeInnerList` this morning, will try it later today. Thanks for the work.
